src/Rings/mpoly-ideals.jl

```diff
@@ -650,39 +641,43 @@ function _compute_radical(I::T) where {T <: MPolyIdeal}
   return Irad
 end
 
+# Go through the generators of I and replace each generator by its squarefree part. 
+# This is a heuristic preprocessing method for radical computations.
+function _squarefree_generator_ideal(I::MPolyIdeal)
+  R = base_ring(I)
+  res_gens = elem_type(R)[]
+  for g in gens(I)
+    is_zero(g) && continue
+    # In characteristic zero, there are special methods, but not in pos. char. 
+    # See e.g. Geddes et. al. "Algorithms for Computer Algebra", Springer (1992)
+    fact = is_zero(characteristic(R)) ? factor_squarefree(g) : factor(g)
+    is_empty(fact) && return ideal(R, one(R))
+    h = prod(x for (x, _) in fact; init=one(R))
+    push!(res_gens, h)
+  end
+  return ideal(R, res_gens)
+end
+
 # Rerouting via expansion of the coefficient field
 function radical(
     I::MPolyIdeal{T};
-    factor_generators::Bool=true, 
```

Comment:
As we are still using factorization and, from what I hear from @fieker and @thofma this is even the way to go, we could think of keeping the old keyword (with an eye towards not introducing breaking changes?).